### DISCUSSION TITLE
Turn Off Yellow Script Outline

### DISCUSSION THREAD

**Discussion by bobby-sills** [ScratchAddons/ScratchAddons#5649] [Ideas]
The yellow script outline of blocks when they're run is on by default in Scratch. On my display it makes the text a little bit blurry, so I would like the option to turn the outline off.

**Comment by mxmou** [comment]
I think it would be better to try to fix the blurry text.

**Reply by BroJac5246** [reply]
This is an [upstream issue](https://scratch.mit.edu/discuss/topic/560413/), which we should keep in mind.

**Reply by WorldLanguages** [reply]
We could try to fix the blurry text, but clarify that it's a vanilla Scratch bug. Or a browser bug?

**Reply by DNin01** [reply]
I have no idea if it's a bug or not, but I think it could have something to do with the browser trying to be efficient when rendering the glowing blocks. One way to solve this could be to stack non-glowing blocks on top of the glowing blocks.

**Reply by BroJac5246** [reply]
> We could try to fix the blurry text, but clarify that it's a vanilla Scratch bug. Or a browser bug?

It _sounds_ like a browser bug, but it's probably possible for Scratch to fix it. Don't quote me on that, though.

> I have no idea if it's a bug or not

Making the text blurry is not at all useful, so it's pretty safe to assume it's a bug.

**Reply by WorldLanguages** [reply]
It's weird that this same bug is happening in Chrome, Firefox AND Safari. So weird.

**Reply by CST1229** [reply]
> It's weird that this same bug is happening in Chrome, Firefox AND Safari. So weird.

Last time I checked, running block text wasn't blurry in Firefox.

**Reply by CST1229** [reply]
Recreating the glow with the `drop-shadow()` filter might work (vanilla glow on the left, my glow on the right):
![](https://user-images.githubusercontent.com/68464103/213860217-817469a7-3542-429b-9618-f5b32f2a72a3.png)
```css
.blocklyDraggable[filter*="url(#blocklyStackGlowFilter"] {
    --radius: 4px;
    --radius-neg: calc(0px - var(--radius));
    --blur: 2px;
    filter:
        drop-shadow(0 0 var(--blur) #FFF200)
        drop-shadow(var(--radius-neg) 0 #FFF200)
        drop-shadow(var(--radius) 0 #FFF200)
        drop-shadow(0 var(--radius-neg) #FFF200)
        drop-shadow(0 var(--radius) #FFF200);
}```

**Reply by DNin01** [reply]
> > I have no idea if it's a bug or not
> 
> 
> 
> Making the text blurry is not at all useful, so it's pretty safe to assume it's a bug.

I guess it is a bug, actually, since this problem was probably unintentional. Could be classified as a regression.

**Reply by DNin01** [reply]
For me, applying any filter at all makes the blocks blurry.

<sub>Microsoft Edge Version 108.0.1462.54 (Official build) (64-bit) / Windows 10 21H2</sub>

**Reply by DNin01** [reply]
One way to potentially fix this might be to visually create a duplicate of the blocks in place without the glow effect, so there would be the same stack of blocks rendering twice, one with a glow effect and one above it without the effect.

![A sharp diamond shape on top of a blurry diamond shape](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/16f9f24b-e3f1-4af2-bef3-961a5b82ad76)

**Reply by BroJac5246** [reply]
> One way to potentially fix this might be to visually create a duplicate of the blocks in place without the glow effect, so there would be the same stack of blocks rendering twice, one with a glow effect and one above it without the effect.
> 
> ![A sharp diamond shape on top of a blurry diamond shape](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/16f9f24b-e3f1-4af2-bef3-961a5b82ad76)
> 

That sounds really inefficient, though.

**Reply by WorldLanguages** [reply]
> For me, applying any filter at all makes the blocks blurry.

Crazy. I wonder what knowledge is required to debug this. Could you test Firefox in the same computer?

**Reply by DNin01** [reply]
> > For me, applying any filter at all makes the blocks blurry.
> 
> Crazy. I wonder what knowledge is required to debug this. Could you test Firefox in the same computer?

Sorry, I don't have Firefox installed.
